File: packages/ts/staffkm-sdk/src/resources/knowledge.ts
import { AxiosInstance } from 'axios'

class KBs {
  constructor(private http: AxiosInstance) {}
  async list(): Promise<any[]> {
    const r = await this.http.get('/api/v1/knowledge')
    return r.data?.data ?? []
  }
  async create(name: string, description?: string): Promise<any> {
    const r = await this.http.post('/api/v1/knowledge', { name, description: description ?? '' })
    return r.data?.data ?? r.data
  }
}

class Docs {
  constructor(private http: AxiosInstance) {}
  /**
   * Upload a document. ``file`` may be a Buffer (Node) or Blob (browser).
   */
  async upload(kbId: string, file: any, filename: string): Promise<any> {
    // FormData differs between Node and browser; let caller pass a ready one or use this helper.
    const FD: any = (globalThis as any).FormData
    if (!FD) throw new Error('FormData not available in this runtime; build a multipart body manually')
    const fd = new FD()
    fd.append('file', file, filename)
    const r = await this.http.post(`/api/v1/knowledge/${kbId}/documents`, fd, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
    return r.data?.data ?? r.data
  }
}

export class KnowledgeResource {
  kbs: KBs
  docs: Docs
  constructor(private http: AxiosInstance) {
    this.kbs = new KBs(http)
    this.docs = new Docs(http)
  }

  async hitTest(kbId: string, query: string, topK = 5): Promise<any[]> {
    const r = await this.http.post(`/api/v1/knowledge/${kbId}/hit_test`, { query, top_k: topK })
    return r.data?.data ?? []
  }
}

File: packages/python/staffkm-sdk/staffkm/resources/knowledge.py
"""Knowledge base resource."""
from __future__ import annotations

from typing import IO
import httpx


class _KBs:
    def __init__(self, http: httpx.Client):
        self._http = http

    def list(self) -> list[dict]:
        r = self._http.get("/api/v1/knowledge")
        r.raise_for_status()
        return r.json().get("data") or []

    def create(self, name: str, description: str | None = None) -> dict:
        r = self._http.post("/api/v1/knowledge", json={
            "name": name,
            "description": description or "",
        })
        r.raise_for_status()
        return r.json().get("data") or r.json()


class _Docs:
    def __init__(self, http: httpx.Client):
        self._http = http

    def upload(self, kb_id: str, file: IO[bytes], filename: str) -> dict:
        files = {"file": (filename, file)}
        r = self._http.post(f"/api/v1/knowledge/{kb_id}/documents", files=files)
        r.raise_for_status()
        return r.json().get("data") or r.json()


class KnowledgeResource:
    def __init__(self, http: httpx.Client):
        self._http = http
        self.kbs = _KBs(http)
        self.docs = _Docs(http)

    def hit_test(self, kb_id: str, query: str, top_k: int = 5) -> list[dict]:
        r = self._http.post(f"/api/v1/knowledge/{kb_id}/hit_test", json={
            "query": query,
            "top_k": top_k,
        })
        r.raise_for_status()
        return r.json().get("data") or []
